classes/class-ppc-pagesetups.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * Pre-Publish Checklist page setups comment
+ *
+ * PHP version 7
+ *
+ * @category PHP
+ * @package  Pre-Publish Checklist.
+ * @author   Display Name <username@BrainstormForce.com>
+ * @license  http://brainstormforce.com
+ * @link     http://brainstormforce.com
+ */
+
+/*
+ * Main Frontpage.
+ *
+ * @since  1.0.0
+ * @return void
+ */
+
+if ( ! class_exists( 'PPC_Pagesetups' ) ) :
+	/**
+	 * Pre-Publish Checklist Loader Doc comment
+	 *
+	 * PHP version 7
+	 *
+	 * @category PHP
+	 * @package  Pre Publish Check-list
+	 * @author   Display Name <username@BrainstormForce.com>
+	 * @license  http://brainstormforce.com
+	 * @link     http://brainstormforce.com
+	 */
+	class PPC_Pagesetups {
+
+		/**
+		 * Member Variable
+		 *
+		 * @var instance
+		 */
+		private static $instance;
+
+		/**
+		 *  Initiator
+		 */
+		public static function get_instance() {
+			if ( ! isset( self::$instance ) ) {
+				self::$instance = new self();
+			}
+			return self::$instance;
+		}
+		/**
+		 * Constructor
+		 */
+		public function __construct() {
+			add_action( 'add_meta_boxes', array( $this, 'ppc_add_custom_meta_box' ) );
+			add_action( 'admin_menu', array( $this, 'ppc_settings_page' ) );
+			add_action( 'wp_ajax_ppc_ajax_add_change', array( $this, 'ppc_meta_box_ajax_add_handler' ), 1 );
+			add_action( 'wp_ajax_nopriv_ppc_ajax_add_change', array( $this, 'ppc_meta_box_ajax_add_handler' ), 1 );
+			add_action( 'wp_ajax_ppc_ajax_delete_change', array( $this, 'ppc_meta_box_ajax_delete_handler' ), 1 );
+			add_action( 'wp_ajax_nopriv_ppc_ajax_delete_change', array( $this, 'ppc_meta_box_ajax_delete_handler' ), 1 );
+			add_action( 'admin_footer', array( $this, 'ppc_markup' ) );
+		}
+		/**
+		 * Function for HTML markup of notification.
+		 *
+		 * Shows the pop-up of warning a user or preventing a user.
+		 *
+		 * @since 1.0.0
+		 */
+		public function ppc_markup() {
+			$ppc_screen = get_current_screen();
+			// If not edit or add new page, post or custom post type window then return.
+			if ( ! isset( $ppc_screen->parent_base ) || ( isset( $ppc_screen->parent_base ) && 'edit' !== $ppc_screen->parent_base ) ) {
+				return;
+			}
+			wp_enqueue_script( 'ppc_backend_checkbox_js' );
+			wp_enqueue_style( 'ppc_backend_css' );
+			?>
+			<div class = "ppc-modal-warn">
+				<div id="ppc_notifications" class="ppc-popup-warn">
+					<h2>Pre-Publish Checklist</h2>
```

Comment:
Make sure all the test in the plugin UI is translatable. [Here is](https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/) more information on this.